tutorial/lesson-3/README.md

```diff
@@ -0,0 +1,321 @@
+these is reflected in a Ruby class that has methods that build and provide the
+template for each display.
+
+Templates are nothing more than HTML with certain elements labeled with an
+`id` that is used by the JavaScript to populate the field with values from
+data from the Controller. The Ruby classes provide that HTML once when the
+JavaScript requests it. It is not generated everytime using ERB. The data
+Controllers, such as the Entry controller, provides only the data, not
+display. The display is left up to the JavaScript.
+
+HTML templates are provided to the JavaScript but one more bit of information
+is needed is needed. Transitions between displays occur when a button is
+pressed. This transition information is included with the templates that are
+passed from the UI controller to the JavaScript.
+
+## Approach
+
+To changes the displays and the transitions to remove the delete option the
+flow for the entry must be changed along with the view, update, and list
+displays. That requires modifying or subclassing the UI classes. Looking at
+the `lib/ui_controller.rb` file note that a `WAB::UI::RestFlow` is created for
+the Entry. This will be replaced with a custom Flow class.
+
+The Flow class creates instances of individual display classes so the Entry
+Flow should create custom classes for each display that remove the delete
+functionality. This lesson will follow my personal preference to keep things
+working and make small incremental change, testing after each change. This
+starts with creating a set of stubs for the classes that will be created.
+
+Not that the reference UI implemenation uses straight HTML and not ERB. This
+is explained on the page [Why Not ERB](erb.md).
+
+## Stubs
+
+Four files will be created in the `lib` directory, `entry_flow.rb`,
+`entry_view.rb`, `entry_update.rb`, and `entry_list.rb`.
+
+`lib/entry_flow.rb`
+
+```ruby
+require 'wab/ui'
+require 'entry_list'
+require 'entry_view'
+require 'entry_update'
+
+class EntryFlow < WAB::UI::RestFlow
+
+  def initialize(shell, template, list_paths)
+    super(shell, template, list_paths)
+  end
+end
+```
+
+`lib/entry_view.rb`
+
+```ruby
+require 'wab/ui'
+
+class EntryView < WAB::UI::View
+
+  def initialize(template, transitions)
+    super('Entry', 'Entry.view', template, transitions, 'ui.View')
+  end
+end
+```
+
+`lib/entry_update.rb`
+
+```ruby
+require 'wab/ui'
+
+class EntryUpdate < WAB::UI::Update
+      
+  def initialize(template, transitions)
+    super('Entry', 'Entry.update', template, transitions)
+  end
+end
+```
+
+`lib/entry_list.rb`
+
+```ruby
+require 'wab/ui'
+
+class EntryList < WAB::UI::List
+
+  def initialize(template, list_paths, transitions)
+    super('Entry', 'Entry.list', template, list_paths, transitions)
+  end
+end
+```
+
+The final step is to switch the `ui_controller.rb` to use the new
+classes. Change the `add_flow` line as shown.
+
+```ruby
+    add_flow(EntryFlow.new(shell,
+```
```

Comment:
This code block is a dead-end. It doesn't clarify anything and moreover, the Ruby expression itself is incomplete.
I recommend that you add the complete *expression* or use a `diff` block (instead of `ruby`) to show what got deleted and what got added..

Reply:
ok, was trying to reduce space. If it isn't clear will put in more.

Reply:
Hows this..